includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,546 @@
+		 *         @type string         $slug           URL of theme preview.
+		 *         @type string         $screenshot_url The URL of a mobile screenshot. Note: if this is empty, the theme may not display.
+		 *         @type string         $homepage        A link to a page with more information about the theme.
+		 *         @type string         $description     A description of the theme.
+		 *         @type string|boolean $requires        Minimum version of WordPress required by the theme. False if all versions are supported.
+		 *         @type string|boolean $requires_php    Minimum version of PHP required by the theme. False if all versions are supported.
+		 *         @type string         $download_link   A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = apply_filters( 'amp_reader_themes', $themes );
+
+		foreach ( $themes as &$theme ) {
+			$theme['availability'] = $this->get_theme_availability( $theme );
+		}
+
+		$this->themes = $themes;
+
+		return $this->themes;
+	}
+
+	/**
+	 * Gets a reader theme by slug.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $slug Theme slug.
+	 * @return array Theme data.
+	 */
+	public function get_reader_theme( $slug ) {
+		return current(
+			array_filter(
+				$this->get_themes(),
+				static function( $theme ) use ( $slug ) {
+					return $theme['slug'] === $slug;
+				}
+			)
+		);
+	}
+
+	/**
+	 * Retrieves theme data.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param boolean $from_api Whether to return theme data from the wordpress.org API. Default false.
+	 * @return array Theme ecosystem posts copied the amp-wp.org website.
+	 */
+	public function get_default_supported_reader_themes( $from_api = false ) {
+		// Note: This can be used to refresh the hardcoded raw theme data.
+		if ( $from_api ) {
+			if ( ! function_exists( 'themes_api' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/theme.php';
+			}
+
+			$response = themes_api(
+				'query_themes',
+				[
+					'author'   => 'wordpressdotorg',
+					'per_page' => 24, // There are only 12 as of 05/2020.
+				]
+			);
+
+			if ( ! $response || is_wp_error( $response ) ) {
+				return $response;
+			}
+
+			if ( is_array( $response ) ) {
+				$response = (object) $response;
+			}
+
+			$supported_themes = array_diff(
+				AMP_Core_Theme_Sanitizer::get_supported_themes(),
+				[ 'twentyten' ] // Excluded because not responsive.
+			);
+
+			$supported_themes_from_response = array_filter(
+				$response->themes,
+				static function( $theme ) use ( $supported_themes ) {
+					return in_array( $theme->slug, $supported_themes, true );
+				}
+			);
+
+			return $supported_themes_from_response;
+		}
+
+		$themes = self::get_default_raw_reader_themes();
+
+		return $themes;
+	}
+
+	/**
+	 * Prepares a single theme.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param array $theme Theme data from the wordpress.org themes API.
+	 * @return array Prepared theme array.
```

Comment:
```suggestion
	 * @return array|object Prepared theme array.
```

See comment on line 210 below.